### ISSUE TITLE
Centos build issue

### ISSUE THREAD

**Issue by sodd** [izderadicka/adecaptcha#1]
Hi, I have trouble to get this work at centos 7.
when I run `python setup.py build_ext --inplace` this warning ocure.

```
running build_ext
building 'adecaptcha.pwrspec' extension
creating build
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/adecaptcha
gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python2.7 -c adecaptcha/pwrspec.c -o build/temp.linux-x86_64-2.7/adecaptcha/pwrspec.o
In file included from /usr/include/numpy/ndarraytypes.h:1781:0,
                 from /usr/include/numpy/ndarrayobject.h:18,
                 from /usr/include/numpy/arrayobject.h:4,
                 from adecaptcha/pwrspec.c:250:
/usr/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^
gcc -pthread -shared -Wl,-z,relro build/temp.linux-x86_64-2.7/adecaptcha/pwrspec.o -L/usr/lib64 -lpython2.7 -o /opt/adecaptcha/adecaptcha/pwrspec.so

```

but it's only warning so i tried next steps and finally when i tried 
`python adecaptcha.py ulozto.cfg http://sound_url` with this result

```
WARNING: pyao module missing, will not be able to play audio
Traceback (most recent call last):
  File "adecaptcha.py", line 41, in main
    res=clslib.classify_audio_file(resp, args[0], ext=get_ext(resp.info()))
  File "/opt/adecaptcha/adecaptcha/clslib.py", line 42, in classify_audio_file
    features= audiolib.calc_mfcc(s, sr, nbins=cfg['nbins'])
  File "/opt/adecaptcha/adecaptcha/audiolib.py", line 259, in calc_mfcc
    rs= mymfcc.mfcc(sample, sr, nbins)
  File "/opt/adecaptcha/adecaptcha/mymfcc.py", line 18, in mfcc
    pw=pwrspec(sig, sr, nbins)
  File "adecaptcha/pwrspec.pyx", line 45, in adecaptcha.pwrspec.pwrspec (adecaptcha/pwrspec.c:2825)
  File "adecaptcha/pwrspec.pyx", line 21, in adecaptcha.pwrspec.winspos (adecaptcha/pwrspec.c:2221)
  File "adecaptcha/pwrspec.pyx", line 17, in adecaptcha.pwrspec.posbins (adecaptcha/pwrspec.c:2039)
  File "/usr/lib64/python2.7/site-packages/numpy/core/fromnumeric.py", line 2768, in around
    return round(decimals, out)
TypeError: ufunc 'rint' output (typecode 'd') could not be coerced to provided output parameter (typecode 'l') according to the casting rule ''same_kind''

```

Could you pls help me or could you give me some hint? dekuji

**Comment by izderadicka** [comment]
Warning should be fine. 
I guess problem is with this command 
python adecaptcha.py ulozto.cfg http://sound_url

URL should be real link to ulozto.cz audio captcha - e.g. wav file. Test link in browser if it's real audio captcha.

Code is beta/ PoC quality so it's not very resistant to incorrect inputs.

But generally should work I'm using it occasionally without problems.

On March 6, 2016 7:06:40 PM GMT+01:00, sodd notifications@github.com wrote:

> Hi, I have trouble to get this work at centos 7.
> when I run `python setup.py build_ext --inplace` this warning
> ocure.
> 
> ```
> running build_ext
> building 'adecaptcha.pwrspec' extension
> creating build
> creating build/temp.linux-x86_64-2.7
> creating build/temp.linux-x86_64-2.7/adecaptcha
> gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall
> -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong
> --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic
> -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall
> -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong
> --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic
> -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python2.7 -c
> adecaptcha/pwrspec.c -o
> build/temp.linux-x86_64-2.7/adecaptcha/pwrspec.o
> In file included from /usr/include/numpy/ndarraytypes.h:1781:0,
>                 from /usr/include/numpy/ndarrayobject.h:18,
>                 from /usr/include/numpy/arrayobject.h:4,
>                 from adecaptcha/pwrspec.c:250:
> /usr/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning
> "Using deprecated NumPy API, disable it by " "#defining
> NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
> #warning "Using deprecated NumPy API, disable it by " \
>  ^
> gcc -pthread -shared -Wl,-z,relro
> build/temp.linux-x86_64-2.7/adecaptcha/pwrspec.o -L/usr/lib64
> -lpython2.7 -o /opt/adecaptcha/adecaptcha/pwrspec.so
> 
> ```
> 
> but it's only warning so i tried next steps and finally when i tried 
> `python adecaptcha.py ulozto.cfg http://sound_url` with this
> result
> 
> ```
> WARNING: pyao module missing, will not be able to play audio
> Traceback (most recent call last):
>  File "adecaptcha.py", line 41, in main
> res=clslib.classify_audio_file(resp, args[0], ext=get_ext(resp.info()))
> File "/opt/adecaptcha/adecaptcha/clslib.py", line 42, in
> classify_audio_file
>    features= audiolib.calc_mfcc(s, sr, nbins=cfg['nbins'])
>  File "/opt/adecaptcha/adecaptcha/audiolib.py", line 259, in calc_mfcc
>    rs= mymfcc.mfcc(sample, sr, nbins)
>  File "/opt/adecaptcha/adecaptcha/mymfcc.py", line 18, in mfcc
>    pw=pwrspec(sig, sr, nbins)
> File "adecaptcha/pwrspec.pyx", line 45, in adecaptcha.pwrspec.pwrspec
> (adecaptcha/pwrspec.c:2825)
> File "adecaptcha/pwrspec.pyx", line 21, in adecaptcha.pwrspec.winspos
> (adecaptcha/pwrspec.c:2221)
> File "adecaptcha/pwrspec.pyx", line 17, in adecaptcha.pwrspec.posbins
> (adecaptcha/pwrspec.c:2039)
> File "/usr/lib64/python2.7/site-packages/numpy/core/fromnumeric.py",
> line 2768, in around
>    return round(decimals, out)
> TypeError: ufunc 'rint' output (typecode 'd') could not be coerced to
> provided output parameter (typecode 'l') according to the casting rule
> ''same_kind''
> 
> ```
> 
> Could you pls help me or could you give me some hint? dekuji
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/izderadicka/adecaptcha/issues/1

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sodd** [comment]
nah, sorry for bad description. I run this command`python adecaptcha.py ulozto.cfg http://xapca3.uloz.to/4958ec42f3f3ff5ae202871f27c8e96f355e4975/sound.wav` inside `/adecaptcha/adecaptcha` folder

**Comment by izderadicka** [comment]
What are your numpy and scipy versions?

On March 6, 2016 8:03:33 PM GMT+01:00, sodd notifications@github.com wrote:

> nah, sorry for bad description. I run this command`python
> adecaptcha.py ulozto.cfg
> http://xapca3.uloz.to/4958ec42f3f3ff5ae202871f27c8e96f355e4975/sound.wav`
> inside `/adecaptcha/adecaptcha` folder
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/izderadicka/adecaptcha/issues/1#issuecomment-192963436

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sodd** [comment]
```
In [2]: numpy.version.version
Out[2]: '1.10.4'
```

```
In [4]: scipy.version.version
Out[4]: '0.12.1'
```

**Comment by izderadicka** [comment]
Should be fixed now in latest commit 077d219d48dc7b16af4bad0936f56f301ce187d2.   Looks like new numpy does not like some implicit type conversions.
